src/dust/coagulation/coagulation.cpp

```diff
@@ -0,0 +1,308 @@
+//========================================================================================
+// (C) (or copyright) 2023-2024. Triad National Security, LLC. All rights reserved.
+//
+// This program was produced under U.S. Government contract 89233218CNA000001 for Los
+// Alamos National Laboratory (LANL), which is operated by Triad National Security, LLC
+// for the U.S. Department of Energy/National Nuclear Security Administration. All rights
+// in the program are reserved by Triad National Security, LLC, and the U.S. Department
+// of Energy/National Nuclear Security Administration. The Government is granted for
+// itself and others acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+// license in this material to reproduce, prepare derivative works, distribute copies to
+// the public, perform publicly and display publicly, and to permit others to do so.
+//========================================================================================
+
+// Artemis includes
+#include "dust/coagulation/coagulation.hpp"
+#include "artemis.hpp"
+#include "dust/dust.hpp"
+#include "geometry/geometry.hpp"
+
+using namespace parthenon::package::prelude;
+
+namespace Dust {
+namespace Coagulation {
+//----------------------------------------------------------------------------------------
+//! \fn  StateDescriptor Coagulalation::Initialize
+//! \brief Adds intialization function for coagulation package
+std::shared_ptr<StateDescriptor> Initialize(ParameterInput *pin) {
+  auto coag = std::make_shared<StateDescriptor>("coagulation");
+  Params &params = coag->AllParams();
+
+  CoagParams cpars;
+  const int nm = pin->GetOrAddInteger("dust", "nspecies", 1);
+  cpars.nm = nm;
+  cpars.vfrag = pin->GetOrAddReal("dust", "vfrag", 1.e3); // cm/s
+  cpars.nlim = 1e10;
+  cpars.integrator = pin->GetOrAddInteger("dust", "coag_int", 3);
+  cpars.use_adaptive = pin->GetOrAddBoolean("dust", "coag_use_adaptiveStep", true);
+  cpars.mom_coag = pin->GetOrAddBoolean("dust", "coag_mom_preserve", true);
+  cpars.nCall_mx = pin->GetOrAddInteger("dust", "coag_nsteps_mx", 1000);
+  // dust particle internal density g/cc
+  const Real rho_p = pin->GetOrAddReal("dust", "grain_density", 1.25);
```

Comment:
We need to be careful about adding new default values to parameters that have already been read by the dust package. `nspecies`, `grain_density` are two examples.

Reply:
I will modify them to use the same value.

Reply:
fixed in the new updated code